### PR TITLE
Add wrapper and class for anon requests

### PR DESCRIPTION
**Pull request by Saundr21**
This PR adds in a new class to process requests through the Anon Client

It is a wrapper around axios such that the user does not need to setup the SoxyProxyAgent themselves.

Getters have also been added to the main Anon class.

### CODE REVIEW

File: src/anon-request.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { Anon } from './anon';
+
+export class AnonRequest {
```
Comment (yumirkov) [inline src/anon-request.ts:5]: Better to call it AnonSocksClient or smth like this

File: src/anon-request.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    this.agent = this.createAgent();
+    this.axios = axios.create({
+      httpAgent: this.agent,
+      httpsAgent: this.agent
+    });
+  }
+
+  private createAgent(): SocksProxyAgent {
+    const socksPort = this.anon.getSOCKSPort() || 9050;
+    return new SocksProxyAgent(`socks://127.0.0.1:${socksPort}`);
```
Comment (yumirkov) [inline src/anon-request.ts:21]: move hardcode to property

File: src/anon-socks-client.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    
+    this.agent = this.createAgent();
+    this.axios = axios.create({
+      httpAgent: this.agent,
+      httpsAgent: this.agent
+    });
+  }
+
+  private createAgent(): SocksProxyAgent {
+    return new SocksProxyAgent(`socks://127.0.0.1:${this.socksPort}`);
```
Comment (yumirkov) [inline src/anon-socks-client.ts:33]: move host to property to be able to configure it in the future or now, if not localhost

File: src/anon-socks-client.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  constructor(anonOrPort: Anon | number) {
+    if (anonOrPort instanceof Anon) {
+      this.anon = anonOrPort;
+      this.socksPort = this.anon.getSOCKSPort();
+    } else {
+      this.socksPort = anonOrPort;
+    }
+
+    if (this.socksPort === 0) {
+      throw new Error('Invalid SOCKS port: 0. SOCKS proxy must be enabled with a valid port number (1-65535).');
```
Comment (yumirkov) [inline src/anon-socks-client.ts:22]: 0 is valid, but not possible to use it for such purpose. Maybe change the message

File: src/anon.ts [hunk 5]
```diff
@@ -7,13 +7,32 @@ import chalk from 'chalk';
  * Allows to run Anon client with different configuration options
  */
 export class Anon {
-  private options?: AnonConfig;
+  private options: AnonConfig = {
+    displayLog: false,
+    useExecFile: false,
+    socksPort: 9050,
+    orPort: 9001,
```
[the remark this answers is not in the document]
Comment (Saundr21) [reply src/anon.ts:14]: Anonrc defaults are 9050 for the socksPort and 0 for the orPort so we are changing to that

**Review by yumirkov — APPROVED** [review]

**Pull request merged**